pkg/network/vmispec/interface_test.go

```diff
@@ -270,6 +270,105 @@ var _ = Describe("VMI network spec", func() {
 			Expect(netvmispec.BindingPluginNetworkWithDeviceInfoExist(ifaces, bindingPlugins)).To(BeTrue())
 		})
 	})
+
+	Context("IndexPodIfaceNamesByNetworkName", func() {
+		const (
+			primaryNetworkName  = "default"
+			primaryPodIfaceName = "eth0"
+
+			secondaryNetworkName1  = "secondary-net1"
+			secondaryPodIfaceName1 = "pod12345"
+
+			secondaryNetworkName2  = "secondary-net2"
+			secondaryPodIfaceName2 = "pod67891"
+		)
+
+		It("Should return an empty map when there are no networks an no interface statuses", func() {
+			res, err := netvmispec.IndexPodIfaceNamesByNetworkName([]v1.Network{}, []v1.VirtualMachineInstanceNetworkInterface{})
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
nit: This is not needed if you do this:

```go
Expect(netvmispec.IndexPodIfaceNamesByNetworkName([]v1.Network{}, []v1.VirtualMachineInstanceNetworkInterface{})).To(BeEmpty())
```

The error check is implicit.

I see this is possible to make use of this in other assertions below.
Having direct assertions with less variables is easier to read IMO, but that may be opinionated so it is not a must.